### DISCUSSION TITLE
Change Optimizer when validation loss stops improving

### DISCUSSION THREAD

**Discussion by athenawisdoms** [Lightning-AI/pytorch-lightning#6796] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Hello, how can we configure Lightning to switch from one optimizer to the second optimizer (e.g. Adam to SGD) when the validation loss stops improving for a certain number of validation epochs? 

Kind of like `EarlyStopping` and its patience parameter, but switches optimizer instead of stop training.

Thank you!

**Comment by awaelchli** [comment]
Hi
This is not so easy, because the optimizer configuration setup depends on the training plugins being used. 
It would be a useful feature though. 
Could you file a feature request on the issues page here on github?

**Reply by athenawisdoms** [reply]
Thanks! Opened issue https://github.com/PyTorchLightning/pytorch-lightning/issues/6829
